.github/workflows/client-android-eas.yml

```diff
@@ -0,0 +1,103 @@
+    # Build a versioned client
+    - cron: '20 5 * * 1,3,5'
+    # 5:20 AM UTC time on every Monday
+    # Run versioning process for the next sdk and build a versioned client
+    - cron: '20 5 * * 1'
+  pull_request:
+    paths:
+      - .github/workflows/client-android-eas.yml
+      - apps/eas-expo-go/**
+
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: 👀 Checkout
+        uses: actions/checkout@v3
+        with:
+          submodules: true
+      - name: 🔓 Decrypt secrets if possible
+        uses: ./.github/actions/expo-git-decrypt
+        with:
+          key: ${{ secrets.GIT_CRYPT_KEY_BASE64 }}
+      - name: ➕ Add `bin` to GITHUB_PATH
+        run: echo "$(pwd)/bin" >> $GITHUB_PATH
+      - name: ♻️ Restore caches
+        uses: ./.github/actions/expo-caches
+        id: expo-caches
+        with:
+          yarn-workspace: 'true'
+          yarn-tools: 'true'
+      - name: 🧶 Yarn install
+        if: steps.expo-caches.outputs.yarn-workspace-hit != 'true'
+        run: yarn install --frozen-lockfile
+      - name: Install eas-cli
+        run: npm install -g eas-cli
+      - name: 🔎 Check which flavor to build
+        id: flavor
+        uses: dorny/paths-filter@v2
+        with:
+          # this action fails when base is not set on schedule event
+          base: ${{ github.ref }}
+          filters: |
+            versioned:
+              - android/versioned-abis/**
+              - android/versioned-react-native/**
+              - android/expoview/src/versioned/**
+              - android/expoview/src/main/java/versioned/**
+              - android/**/*.gradle
+      - name: Resolve profile
+        id: profile
+        run: |
+          DISPATCH_PROFILE="${{ github.event.inputs.buildType }}"
+          IS_VERSIONED="${{ steps.flavor.outputs.versioned }}"
+          if [[ ! -z "$DISPATCH_PROFILE" ]]; then
+            echo ::set-output name=profile::"$DISPATCH_PROFILE"
+          elif [[ "${{ github.event.schedule }}" == "20 5 * * 1,3,5" ]]; then
+            echo ::set-output name=profile::"versioned-client"
+          elif [[ "${{ github.event.schedule }}" == "20 5 * * 1" ]]; then
+            echo ::set-output name=profile::"versioned-client-add-sdk"
+          elif [[ "$IS_VERSIONED" == "true" ]]; then
+            echo ::set-output name=profile::"versioned-client"
+          else
+            echo ::set-output name=profile::"unversioned-client"
+          fi
+      - name: Build
+        uses: ./.github/actions/eas-build
+        id: build
+        with:
+          platform: 'android'
+          profile: ${{ steps.profile.outputs.profile  }}
+          projectRoot: './apps/eas-expo-go'
+          expoToken: ${{ secrets.EAS_BUILD_BOT_TOKEN }}
+          resourceClass: large
+          noWait: ${{ github.event.schedule }}
```

Comment:
in case of cron jobs workflow will finish immediately after build is scheduled, I'm wondering whether we should do the same for manually triggered builds

Reply:
yes, that is why I'm doing that only on cronjobs and maybe manually triggered runs where we do no really need those stauses, e.g. if a cron job that builds a client fails we could still send a message to slack, the same way as it is done now, but from eas directlly.

> as an aside - this would be a generally useful feature to have for eas build.

Github provides a checks API that allows us to programmatically create/update statuses displayed under PR, but it can't be used with regular user tokens, you would need to have a github app hosted somewhere to interact with that API. Creating sth like that is a big project and I don't think it makes sense to implement that just for internal usage, but if we want to expose that app to users, then it would justify the additional work

Reply:
i agree, we don't need to implement this for internal usage. @FiberJW and @EvanBacon have been working on expo github integration in general so we may be able to piggyback off of that at some point. i'll just make a task for us to follow up on this in the future

###### <a href="https://getunblocked.com"><img src="https://avatars.githubusercontent.com/in/166219" width=20 align="center" /></a>&nbsp;&nbsp;Comment posted from [Unblocked](https://getunblocked.com/dashboard/team/5bf29d2c-23ae-4af8-8ccc-fe767cd2df79/thread/efb75468-2aca-4cde-9478-10618748be30?message=75b834fd-cc08-4bd3-980e-61b4e234a702).